isthmus/src/main/java/io/substrait/isthmus/calcite/SubstraitOperatorTable.java

```diff
@@ -0,0 +1,73 @@
+package io.substrait.isthmus.calcite;
+
+import io.substrait.isthmus.AggregateFunctions;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.apache.calcite.sql.SqlFunctionCategory;
+import org.apache.calcite.sql.SqlIdentifier;
+import org.apache.calcite.sql.SqlKind;
+import org.apache.calcite.sql.SqlOperator;
+import org.apache.calcite.sql.SqlOperatorTable;
+import org.apache.calcite.sql.SqlSyntax;
+import org.apache.calcite.sql.fun.SqlStdOperatorTable;
+import org.apache.calcite.sql.util.SqlOperatorTables;
+import org.apache.calcite.sql.validate.SqlNameMatcher;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+/**
+ * Overrides SQL operator lookups to return Substrait specific functions variants (e.g. {@link
+ * AggregateFunctions#MAX}} when they are available.
+ */
+public class SubstraitOperatorTable implements SqlOperatorTable {
+
+  public static SubstraitOperatorTable INSTANCE = new SubstraitOperatorTable();
+
+  private static final SqlOperatorTable SUBSTRAIT_OPERATOR_TABLE =
+      SqlOperatorTables.of(
+          List.of(
+              AggregateFunctions.MAX,
+              AggregateFunctions.MIN,
+              AggregateFunctions.AVG,
+              AggregateFunctions.SUM,
+              AggregateFunctions.SUM0));
+
+  // kinds for which Substrait specific operators are provided
+  private static final Set<SqlKind> OVERRIDE_KINDS =
```

Comment:
Couple minor things:
* consider using an [EnumSet](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/EnumSet.html) here
* I wonder if it would be easier to maintain this list if it could be automatically computed somehow from `SUBSTRAIT_OPERATOR_TABLE` - would that be possible?

Reply:
TIL about EnuSet. Done